File: 3658-gcd-of-odd-and-even-sums/3658-gcd-of-odd-and-even-sums.cpp
class Solution {
public:
    int firstApproach(int n){
        int oddSum = 0; 
        int evenSum = 0; 
        bool flag = true; 

        for(int i = 1; i <= (2*n); i++){
            if(flag){
                oddSum += i; 
                flag = false; 
            }else{
                evenSum += i; 
                flag = true; 
            }
        }

        return gcd(oddSum, evenSum); 

    }

    int secondApproach(int n){
        int sumOfEvenNumber = n*(n+1); 
        int sumOfOddNumber = n*n; 
        cout << "Even : " << sumOfEvenNumber << " Odd : " << sumOfOddNumber << endl; 
        return gcd(sumOfOddNumber, sumOfEvenNumber); 
    }

    int gcdOfOddEvenSums(int n) {
        // return firstApproach(n); 
        return secondApproach(n); 
    }
};

File: 3658-gcd-of-odd-and-even-sums/3658-gcd-of-odd-and-even-sums.py
class Solution:
    def gcdOfOddEvenSums(self, n: int) -> int:
        oddSum = 0
        evenSum = 0
        flag = True
        for i in range(1,2*n):
            if flag:
                oddSum = oddSum + i
                flag = False

            else:
                evenSum = evenSum + i
                flag = True
        
        return math.gcd(oddSum, evenSum)